src/components/tooltip/README.md

```diff
@@ -0,0 +1,67 @@
+# Tooltip
+
+#### Description
+
+Tooltips display informative text when users hover over or focus on an element.
+
+#### Usage
+
+```js
+<Tooltip
+  position="left"
+  label="If you buy a pizza, you will also get a free ice cream :)"
+>
+  <button onClick={orderPizza({ freeIceCream: 'yes please' })}>Submit</button>
+</Tooltip>
+```
+
+#### Working with disabled child elements
+
+When you use a tooltip with a disabled element, you need to wrap element in a div, as
+disabled elements do not fire events.
+
+```js
+<Tooltip
+  position="left"
+  label="You do not have permission to delete the database"
+>
+  <div style={{ cursor: 'not-allowed' }}>
```

Comment:
How about we let the tooltip do the wrapping and the consumer can pass custom styles for this wrapper div, in case it's necessary. For example:

```js
<Tooltip
  disabledWrapperStyles={{...}}
  disabledWrapperClassName=""
>
  <button />
</Tooltip>
```

Reply:
Not sure. I'd be as restrictive with UIKit as one can be. Passing `classNames` or `styles` we should maybe avoid for as long as we ever can? Maybe if needed than rather a `renderWrapper` or something to give full control.

Reply:
Either way it's a trade off. I tried with creating a wrapper div (like we do in `core` Tooltip), and I found it's a lot easier for styling purposes if we don't create a wrapper div.

Reply:
The styles/classname prop names are on purpose very specific to the "wrapped element", so it's not that we spread them everywhere.

I just find it a bit weird to have to wrap a div, only when in disabled state. It's going to be a "mess" from a consumer perspective

```js
<Tooltip
>
  {isDisabled ? (
    <div style={{ cursor: 'not-allowed' }}>
      <MyComponent isDisabled={isDisabled} />
    </div>
  ) : (
    <MyComponent />
  )}
</Tooltip>
```

Reply:
@emmenko but then we are just complaining about how the dom / html works. It's not unique to tooltips that disabled elements don't fire events 🤔

Reply:
🤷‍♂️ . If we wrapped in a div, we could have like a "display: block" vs "display: inline-block" option.

Reply:
Whatever works, as long as the API is simple to use.

Reply:
Just leaving my humble opinion here after @montezume explained me both approaches.

I think I'm more with the mindset of Tobi, of not allowing the consumer "insert" custom styles in the component (eventhough this custom style for the wrapping div are not intended for customizing but for consistency I would say).

The reason is because allowing the user to add custom styles is not how the rest of components of UIKit works (maybe just collapsible panel which is quite old). So, for the sake of consistency (API wise) I would vote for just rely on the user to add proper children elements to the `Tooltip`.

If it's well documented, explaining the use case and what you should do to make it consistent in a specific scenario, for me it's more than enough as a consumer 🤷‍♂️ .

Reply:
I would like to keep the API like it is for now, and if we run into problems with the migration, I'm 100% open to reworking it.